src/physlr-molecules.cc

```diff
@@ -314,35 +325,82 @@ biconnectedComponents(graph_t& subgraph, vertexToComponent_t& vertexToComponent)
 	}
 }
 
+template<class Graph, class vertexIter, class edgeSet>
+void
+make_subgraph(Graph& g, Graph& subgraph, edgeSet& edge_set, vertexIter vBegin, vertexIter vEnd)
+{
+	// //   Make a vertex-induced subgraph of graph g, based on vertices from vBegin to vEnd
+	// //   track the source node by indexOriginal
+
+	// Add vertices into the subgraph, but set `indexOriginal` the index of it in the source graph.
+	for (auto& vIter = vBegin; vIter != vEnd; ++vIter) {
+		auto u = boost::add_vertex(subgraph);
+
+		subgraph[u].name = g[*vIter].name;
+		subgraph[u].weight = g[*vIter].weight;
+		subgraph[u].indexOriginal = g[*vIter].indexOriginal;
+	}
+
+	// Iterate over all pairs of vertices in the subgraph:
+	// check whether there exist an edge between their corresponding vertices in the source graph.
+	graph_t::vertex_iterator vIter1, vIter2, vend1, vend2;
+	for (boost::tie(vIter1, vend1) = vertices(subgraph); vIter1 != vend1; ++vIter1) {
+		for (boost::tie(vIter2, vend2) = vertices(subgraph); vIter2 != vend2; ++vIter2) {
+			if (vIter1 != vIter2) {
+				auto got = edge_set.find(std::make_pair(
+				    subgraph[*vIter1].indexOriginal, subgraph[*vIter2].indexOriginal));
+
+				if (got != edge_set.end()) {
+					auto new_edge = boost::add_edge(*vIter1, *vIter2, subgraph).first;
+					subgraph[new_edge].weight = got->second;
+				}
+			}
+		}
+	}
+}
+
 int
 main(int argc, char* argv[])
 {
-
 	auto progname = "physlr-molecules";
 	int optindex = 0;
 	static int help = 0;
 	std::string separationStrategy = "bc";
+	uint64_t threads = 1;
 	bool verbose = false;
 	bool failed = false;
 	static const struct option longopts[] = {
 		{ "help", no_argument, &help, 1 },
 		{ "separation-strategy", required_argument, nullptr, 's' },
 		{ nullptr, 0, nullptr, 0 }
 	};
-	for (int c; (c = getopt_long(argc, argv, "s:v", longopts, &optindex)) != -1;) {
+#if _OPENMP
+	for (int c; (c = getopt_long(argc, argv, "s:vt:", longopts, &optindex)) != -1;) {
+#else
+	for (int c; (c = getopt_long(argc, argv, "s:v:", longopts, &optindex)) != -1;) {
+#endif
```

Comment:
`for (int c; (c = getopt_long(argc, argv, "s:vt:", longopts, &optindex)) != -1;) {` Remove all that and keep this.